src/darwin/Framework/CHIP/MTRDefines_Internal.h

```diff
@@ -46,4 +46,14 @@
     _Pragma("clang diagnostic ignored \"-Wshadow\"")            \
     __strong typeof(local) _Nullable local = _mtr_weak_##local  \
     _Pragma("clang diagnostic pop")
+
+/// Declares an unused local of unspecified type, to prevent accidental
+/// references to a shadowed variable of the same name. Note that hiding
+/// `self` does not prevent implicit references to self due to ivar access.
+#define mtr_hide(local)                                         \
+    _Pragma("clang diagnostic push")                            \
+    _Pragma("clang diagnostic ignored \"-Wshadow\"")            \
+    __attribute__((unused)) _mtr_hidden local;                  \
+    _Pragma("clang diagnostic pop")
+typedef struct {} _mtr_hidden;
```

Comment:
Since `MTR_HIDDEN` exists and means something else, I suggest naming this something like `_mtr_hidden_var`, just so this is clear at a glance.